framework/TApplication.php

```diff
@@ -698,6 +699,60 @@ public function getModules()
 		return $this->_modules;
 	}
 
+	/**
+	 * Returns a list of application modules of a specific class.
+	 * Modules that have not been loaded yet are loaded.
+	 * @param string $type class name of the modules to look for.
+	 * @param bool $strict should just the class be found or subclasses of the class
+	 * @param bool $first return the first module found, default false.
+	 * @return array|object list of loaded application module of a specific class
+	 * or the first object of a specific class
+	 * @since 4.2.0
+	 */
+	public function getModulesByType($type, $strict = false, $first = false)
+	{
+		$m = [];
+		foreach ($this->_modules as $id => $module) {
+			if ($module === null) {
+				[$moduleClass, $initProperties, $configElement] = $this->_lazyModules[$id];
+				
+				$match = ($moduleClass === $type);
```

Comment:
Maybe this could be made easier and avoid the use of reflection?
```php
$match = $strict ? ($moduleClass === $type) : ($moduleclass instanceof $type);
```